woocommerce/woocommerce-gutenberg-products-block
# Create `StoreNoticesWithSnackbarProvider`

This PR will create a new component `StoreNoticesWithSnackbarProvider` which is based on `StoreNoticesProvider`. The snackbar related code is removed from `StoreNoticesProvider`.

Any components that need to manipulate store notices, but not the snackbar (All Products) should be using `StoreNoticesProvider` and any that do need to manipulate the snackbar should use `StoreNoticesWithSnackbarProvider`.


This will resolve #4405 as long as #4406 is merged too.




### How to test the changes in this Pull Request:

1. Run `npm run build:deploy` - inspect `build/all-products.asset.php` and verify `wc-blocks-checkout` is not there.
2. Go to the All products block, find a product that you want to add to the cart, don't add it yet!
3. In a new tab, edit this product so it is out of stock.
4. Go back to the All products block, and try to add the product to your cart.
5. Check you get a store notice at the top of the page/block.
6. Add some in-stock products to the cart.
7. Ensure you've got a useable coupon on your store.
8. Go to the Cart/Checkout block, add a coupon.
9. Ensure the snackbar shows.
